R/utils.R

```diff
@@ -170,3 +170,41 @@ check_connect_version <- function(using_version, minimum_tested_version = "1.8.8
   }
   invisible()
 }
+
+token_hex <- function(n) {
+  raw_to_hex <- function(raw) {
+    paste(as.character(raw), collapse = "")
+  }
+  raw <- as.raw(sample(0:255, n, replace = TRUE))
+  raw_to_hex(raw)
+}
+
+content_is_rendered <- function(app_mode) {
+  if ((app_mode) %in% c("rmd-static", "jupyter-static", "quarto-static")) {
+    TRUE
+  } else {
+    FALSE
+  }
+}
+
+content_is_interactive <- function(app_mode) {
+  interactive_app_modes <- c(
+    "shiny",
+    "rmd-shiny",
+    "jupyter-voila",
+    "python-api",
+    "python-dash",
+    "python-streamlit",
+    "python-bokeh",
+    "python-fastapi",
+    "python-shiny",
+    "quarto-shiny",
+    "tensorflow-saved-model",
+    "api"
+  )
+  if ((app_mode) %in% interactive_app_modes) {
+    TRUE
+  } else {
+    FALSE
+  }
+}
```

Comment:
We don't have to do this here, but it feels a little funny that we use these at our top level and pass the actual app modes to them. Why not make it a method on a piece of content so that you don't have to know the internal structure of the R6 content method to extract the mode and then pull this out?

Reply:
And FWIW, I noticed this not in reading these functions themselves but seeing how we call them up in https://github.com/rstudio/connectapi/pull/283/files#diff-2517925fdf51a96dc6acb93b597db70c334538b5b9f033cbe48e1b17a86d3455R954

Reply:
I thought the exact same thing when I was writing that same code! I'll make that change.